executor/pipeline/fetch.rs

```diff
@@ -316,27 +404,74 @@ fn execute_attributes_list<'a>(
     thing_manager: Arc<ThingManager>,
 ) -> Result<DocumentList, FetchExecutionError> {
     let mut list = DocumentList::new();
-    let mut iter = object.get_has_type_unordered(snapshot.as_ref(), thing_manager.as_ref(), attribute_type);
+    let mut iter = prepare_attribute_type_has_iterator(object, attribute_type.clone(), &snapshot, &thing_manager)?;
+
     while let Some(result) = iter.next() {
-        match result {
-            Ok((attribute, count)) => {
-                for _ in 0..count {
-                    list.list.push(DocumentNode::Leaf(DocumentLeaf::Concept(Concept::Thing(Thing::Attribute(
-                        attribute.as_reference().into_owned(),
-                    )))));
-                }
+        let (has, count) = result.map_err(|source| FetchExecutionError::ConceptRead { source })?;
+        let attribute = has.attribute();
+        let suitable = attribute
+            .type_()
+            .is_subtype_transitive_of_or_same(snapshot.as_ref(), thing_manager.type_manager(), attribute_type.clone())
+            .map_err(|source| FetchExecutionError::ConceptRead { source })?;
+        if suitable {
+            for _ in 0..count {
+                list.list.push(DocumentNode::Leaf(DocumentLeaf::Concept(Concept::Thing(Thing::Attribute(
+                    has.attribute().into_owned(),
+                )))));
             }
-            Err(err) => return Err(FetchExecutionError::ConceptReadError { source: err }),
         }
     }
     Ok(list)
 }
 
+fn prepare_attribute_type_has_iterator<'a>(
+    object: impl ObjectAPI<'a>,
+    attribute_type: AttributeType<'static>,
+    snapshot: &Arc<impl ReadableSnapshot>,
+    thing_manager: &Arc<ThingManager>,
+) -> Result<HasIterator, FetchExecutionError> {
+    let subtypes = attribute_type
+        .get_subtypes_transitive(snapshot.as_ref(), thing_manager.type_manager())
+        .map_err(|source| FetchExecutionError::ConceptRead { source })?;
+    let attribute_types = TypeAPI::chain_types(attribute_type.clone(), subtypes.into_iter().cloned());
+
+    object
+        .get_has_types_range_unordered(snapshot.as_ref(), thing_manager.as_ref(), attribute_types)
+        .map_err(|source| FetchExecutionError::ConceptRead { source })
+}
+
+fn execute_single_function<Snapshot: ReadableSnapshot + 'static>(
```

Comment:
This function doesn't execute though, right? I t creates and seeds the executor.